ConfigSpace/hyperparameters.pyx

```diff
@@ -608,6 +766,20 @@ cdef class UniformFloatHyperparameter(FloatHyperparameter):
                 neighbors.append(neighbor)
         return neighbors
 
+    def _pdf(self, vector: np.ndarray) -> np.ndarray:
+        # everything that comes into _pdf for a uniform variable should
+        # already be in [0, 1]-range, and if not, it's outside the upper
+        # or lower bound.
+        ub = 1
+        lb = 0
+        inside_range = ((lb <= vector) & (vector <= ub)).astype(int)
+        return inside_range / (self.upper - self.lower)
```

Comment:
Fair point. Since not all inverse_transform work the same way (Uniform -> [0, 1], Normal -> [self._lower, self._upper], Beta -> [self._lower, self._upper], _pdf has to work with what it is given. Moreover, all FloatHPs have the same pdf, which forces the _pdf to get the user the expected output.

My approach has been to give the results that would be the most sensible to me if I were a user. As a user, I would expect a Uniform between 0 and 5 to have density 1/5 everywhere, and for that to be the case, _pdf has to look like this.

A similar discussion can be had about what the pdf of a logged uniform should output (arguably, it should be a log-uniform, but it's not). However, it is just uniform in our case, since I think that is what the user will have in mind.